### PR TITLE
Doc: Introduce snmp-integration for Technical Preview

### PR DESCRIPTION
**Pull request by karenzone**
Related: https://github.com/elastic/logstash/issues/16145

We'll need to tweak content and messaging for each branch for expectations and accuracy. 

**PREVIEW:** https://logstash-docs_bk_1648.docs-preview.app.elstc.co/guide/en/logstash/master/plugins-integrations-snmp.html

### CODE REVIEW

File: docs/plugins/integrations/snmp.asciidoc [hunk 1]
```diff
@@ -14,8 +14,123 @@ START - GENERATED VARIABLES, DO NOT EDIT!
...
-[id="plugins-{type}s-{plugin}"]
+// include::{include_path}/plugin_header.asciidoc[]
 
-=== SNMP Integration Plugin
+experimental[]
 
-COMING SOON!
+.Technical Preview
+****
+The new `integration-snmp` plugin, and its component plugins--`input-snmp` and `input-snmptrap`--are available in _Technical Preview_ at {ls} 8.14.0.
```
Comment (karenzone) [inline docs/plugins/integrations/snmp.asciidoc:23]: Note to self @karenzone: Get exact versioning right, and determine how we want to handle this for minor and patch releases (lockfile).

**Comment by karenzone** [comment]
Proposal outlined in the issue: 
https://github.com/elastic/logstash/issues/16145#issuecomment-2099424617

File: docs/plugins/integrations/snmp.asciidoc [hunk 2]
```diff
@@ -18,4 +18,127 @@ END - GENERATED VARIABLES, DO NOT EDIT!
...
-COMING SOON!
+* Plugin version: v4.0.0
+* Released on: 2024-05-03
+* https://github.com/logstash-plugins/logstash-integration-snmp/blob/main/CHANGELOG.md[Changelog]
+
+experimental[]
+
+.Technical Preview
+****
+The new `integration-snmp` plugin, and its component plugins--`input-snmp` and `input-snmptrap`--are available in _Technical Preview_ at {ls} 8.14.0.
```
Comment (jsvd) [inline docs/plugins/integrations/snmp.asciidoc:29]:
```suggestion
The new `integration-snmp` plugin, and its component plugins--`input-snmp` and `input-snmptrap`--are available in _Technical Preview_ and can be installed on the latest Logstash 7.x and 8.x versions.
```

File: docs/plugins/integrations/snmp.asciidoc [hunk 3]
```diff
@@ -18,4 +18,127 @@ END - GENERATED VARIABLES, DO NOT EDIT!
...
+
+Pre-4.0.0 versions of the `input-snmp` plugin are bundled with {ls} by default, and will not be upgraded to the snmp integration without your deliberate action.
```
Comment (jsvd) [inline docs/plugins/integrations/snmp.asciidoc:31]:
```suggestion
Current 1.x versions of the `input-snmp` plugin are bundled with {ls} by default, and will soon be replaced by the snmp input plugin contained in this integration.
```

File: docs/plugins/integrations/snmp.asciidoc [hunk 4]
```diff
@@ -18,4 +18,127 @@ END - GENERATED VARIABLES, DO NOT EDIT!
...
+
+// Uncomment and replace with above bullets after generated the initial versions of the target files
+// - {logstash-ref}/plugins-input-snmp.html[SNMP input plugin]
+// - {logstash-ref}/plugins-input-snmptrap.html[Snmptrap input plugin]
+
+The new the `logstash-integration-snmp` plugin combines the 
+`logstash-input-snmp` and `logstash-input-snmptrap` plugins into one integrated plugin that encompasses
+the capabilities of both. 
+This integrated plugin package provides better alignment in snmp processing, better resource managenment, 
+easier package maintenance, and a smaller installation footprint.
```
Comment (jsvd) [inline docs/plugins/integrations/snmp.asciidoc:53]:
```suggestion
The new `logstash-integration-snmp` plugin combines the 
`logstash-input-snmp` and `logstash-input-snmptrap` plugins into a single Ruby gem.
This integrated plugin package provides better alignment in behaviour when it comes to SNMP processing, better resource management, easier package maintenance, and a smaller installation footprint. 
The individual plugins now share the same code base and have been refactored to leverage the latest version of https://www.snmp4j.org/[SNMP4j].
```

File: docs/plugins/integrations/snmp.asciidoc [hunk 5]
```diff
@@ -18,4 +18,127 @@ END - GENERATED VARIABLES, DO NOT EDIT!
...
+`logstash-input-snmp` and `logstash-input-snmptrap` plugins into one integrated plugin that encompasses
+the capabilities of both. 
+This integrated plugin package provides better alignment in snmp processing, better resource managenment, 
+easier package maintenance, and a smaller installation footprint. 
+
+
+[id="plugins-{type}s-{plugin}-migration"]
+==== Migrating to `logstash-integration-snmp` from individual plugins
+
+You'll retain and expand the functionality of existing stand-alone plugins, but in a more compact, integrated package.
```
Comment (jsvd) [inline docs/plugins/integrations/snmp.asciidoc:59]:
```suggestion
You'll retain the functionality of existing stand-alone plugins, but with a more consistent experience and expanded features. 
```

File: docs/plugins/integrations/snmp.asciidoc [hunk 6]
```diff
@@ -18,4 +18,127 @@ END - GENERATED VARIABLES, DO NOT EDIT!
...
+You'll retain and expand the functionality of existing stand-alone plugins, but in a more compact, integrated package. 
+In this section, we'll note mapping and behavioral changes, and explain how to preserve current behavior if needed.
+
+[IMPORTANT] 
+--
+Uninstall the `logstash-input-snmp` and `logstash-input-snmptrap` plugins _before_ you install the `logstash-integration-snmp` plugin. 
+
+* `bin/logstash-plugin remove logstash-input-snmp` +
+* `bin/logstash-plugin remove logstash-input-snmptrap`
+--
```
Comment (jsvd) [inline docs/plugins/integrations/snmp.asciidoc:68]:
```suggestion
```
This shouldn't be necessary as installing the snmp integration plugin uninstalls the individual plugins it provides.

File: docs/plugins/integrations/snmp.asciidoc [hunk 7]
```diff
@@ -18,4 +18,127 @@ END - GENERATED VARIABLES, DO NOT EDIT!
...
+
+// ToDo: Add guidance and set expectations for when snmp integration is installed by default
+
+[id="plugins-{type}s-{plugin}-migration-input-snmp"]
+===== Migration notes: `logstash-input-snmp` 
+
+As a component of the new `logstash-integration-snmp` plugin, the `logstash-input-snmp` plugin offers the same 
+capabilities as the stand-alone https://github.com/logstash-plugins/logstash-input-snmp[logstash-input-snmp]. 
+
+You might need to address some behavior changes depending on the use-case and how the ingested data is being handled through the pipeline.
```
Comment (jsvd) [inline docs/plugins/integrations/snmp.asciidoc:78]:
```suggestion
Given the underlying code refactor to better align snmp and snmptrap plugin behaviour you might need to address some changes depending on the use-case and how the ingested data is being handled through the pipeline.
```

File: docs/plugins/integrations/snmp.asciidoc [hunk 8]
```diff
@@ -18,4 +18,119 @@ END - GENERATED VARIABLES, DO NOT EDIT!
...
+* SNMPtrap input plugin
+
+// Uncomment and replace with above bullets after generated the initial versions of the target files
+// - {logstash-ref}/plugins-input-snmp.html[SNMP input plugin]
+// - {logstash-ref}/plugins-input-snmptrap.html[Snmptrap input plugin]
+
+The new `logstash-integration-snmp` plugin combines the 
+`logstash-input-snmp` and `logstash-input-snmptrap` plugins into a single Ruby gem.
+This integrated plugin package provides better alignment in behaviour when it comes to SNMP processing, better resource management, easier package maintenance, and a smaller installation footprint. 
+The individual plugins now share the same code base and have been refactored to leverage the latest version of https://www.snmp4j.org/[SNMP4j].
```
Comment (karenzone) [inline docs/plugins/integrations/snmp.asciidoc:52]:
```suggestion
This integrated plugin package provides better alignment in SNMP processing behavior, better resource management, easier package maintenance, and a smaller installation footprint. 
The individual plugins now share the same code base and have been refactored to leverage the latest version of https://www.snmp4j.org/[SNMP4j].
```

File: docs/plugins/integrations/snmp.asciidoc [hunk 9]
```diff
@@ -18,4 +18,119 @@ END - GENERATED VARIABLES, DO NOT EDIT!
...
+
+// ToDo: Add guidance and set expectations for when snmp integration is installed by default
+
+[id="plugins-{type}s-{plugin}-migration-input-snmp"]
+===== Migration notes: `logstash-input-snmp` 
+
+As a component of the new `logstash-integration-snmp` plugin, the `logstash-input-snmp` plugin offers the same 
+capabilities as the stand-alone https://github.com/logstash-plugins/logstash-input-snmp[logstash-input-snmp]. 
+
+Given the underlying code refactor to better align snmp and snmptrap plugin behaviour you might need to address some changes depending on the use-case and how the ingested data is being handled through the pipeline.
```
Comment (karenzone) [inline docs/plugins/integrations/snmp.asciidoc:70]:
```suggestion
Given the underlying code refactor to better align snmp and snmptrap plugin behavior, you might need to address some changes depending on your use case and how the ingested data is being handled through the pipeline.
```

**Comment by karenzone** [comment]
@jsvd, love the changes! Thank you for investing.  I need a 👍 so that I can merge this and move on to the next steps. 

Next up:
- [ ]  Add placeholder docs for 8.14
- [ ] Add placeholder docs for 8.13??? (It seems appropriate to give users advanced notice of changes to come, even messaging as early as current version). 
- [ ] Merge [Update plugins-metadata.json with integration-snmp: set skiplist to false #16137](https://github.com/elastic/logstash/pull/16137) to get docs generating on their own.

In the interim, I'll keep a close eye on the generated docs PRs to be sure that we're not accidentally overwriting this carefully curated messaging for each branch.

**Review by jsvd — APPROVED** [review]
LGTM

**Comment by karenzone** [comment]
8216

**Pull request merged**